### PR TITLE
Add Tool Tip for Bread Crumbs in Comparison View

### LINKED ISSUE #1668: Add Tool Tip for Bread Crumbs in Comparison View
Token Index ranges would be nice (start, end).
Also maybe hinting that the number in the breadcrumb is the length of the match measured in the number of tokens.

### PR DESCRIPTION
**Pull request by Kr0nox**
This PR adds tooltips in breadcrumps.
In addition it adds an export of the start and end index of the tokenlist for each match. This change is backwards compatible with 5.0.0 and earlier.
Tooltip in 5.1.0: 
grafik

Tooltip in 5.0.0:
grafik

---

Moreover it includes a rework of the tooltip component. It now can grow outside of a container that has overflow set to hidden, auto or scroll

Fixes #1668

### CODE REVIEW

File: core/src/main/java/de/jplag/reporting/jsonfactory/ComparisonReportWriter.java [hunk 1]
```diff
@@ -112,16 +112,20 @@ private Match convertMatchToReportMatch(JPlagComparison comparison, de.jplag.Mat
...
         int startColumnSecond = startOfSecond.getColumn();
+        int startTokenSecond = match.startOfSecond();
         int endLineSecond = endOfSecond.getLine();
         int endColumnSecond = endOfSecond.getColumn() + endOfSecond.getLength() - 1;
+        int endTokenSecond = match.endOfSecond();
 
-        return new Match(firstFileName, secondFileName, startLineFirst, startColumnFirst, endLineFirst, endColumnFirst, startLineSecond,
-                startColumnSecond, endLineSecond, endColumnSecond, match.length());
+        return new Match(firstFileName, secondFileName, startLineFirst, startColumnFirst, startTokenFirst, endLineFirst, endColumnFirst,
+                endTokenFirst, startLineSecond, startColumnSecond, startTokenSecond, endLineSecond, endColumnSecond, endTokenSecond, match.length());
```
[the remark this answers is not in the document]
Comment (Kr0nox) [reply core/src/main/java/de/jplag/reporting/jsonfactory/ComparisonReportWriter.java:128]: Was thinking the same, already have made a card for that on the board

[the remark this answers is not in the document]
Comment (tsaglam) [reply core/src/main/java/de/jplag/reporting/jsonfactory/ComparisonReportWriter.java:128]: Yeah, I agree. We could just do I class for token subsequences, and a match just takes two subsequences.

**Comment by Kr0nox** [comment]
There is still one bug that happens on smaller screens where the view becomes scrollable to the right. I will fix this before we merge this

**Comment by Kr0nox** [comment]
change token ranges to token indexes of matches

**Comment by Kr0nox** [comment]
Another bug with the new tooltips is that when the match list is scrolled tooltips get displayed wrong

**Review by tsaglam — APPROVED** [review]

**Pull request merged**